server/routes/contacts/manage/phone/edit/manageContactEditPhoneController.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from 'express'
+import { Page } from '../../../../../services/auditService'
+import { PageHandler } from '../../../../../interfaces/pageHandler'
+import ReferenceCodeType from '../../../../../enumeration/referenceCodeType'
+import ReferenceDataService from '../../../../../services/referenceDataService'
+import { PhoneNumberSchemaType } from '../phoneSchemas'
+import { ContactsService } from '../../../../../services'
+import ReferenceCode = contactsApiClientTypes.ReferenceCode
+import ContactPhoneDetails = contactsApiClientTypes.ContactPhoneDetails
+import GetContactResponse = contactsApiClientTypes.GetContactResponse
+
+export default class ManageContactEditPhoneController implements PageHandler {
+  constructor(
+    private readonly contactsService: ContactsService,
+    private readonly referenceDataService: ReferenceDataService,
+  ) {}
+
+  public PAGE_NAME = Page.MANAGE_CONTACT_EDIT_PHONE_PAGE
+
+  GET = async (
+    req: Request<{ prisonerNumber: string; contactId: string; contactPhoneId: string }>,
+    res: Response,
+  ): Promise<void> => {
+    const { user } = res.locals
+    const { contactId, contactPhoneId } = req.params
+    const contact: GetContactResponse = await this.contactsService.getContact(parseInt(contactId, 10), user)
+    const phone: ContactPhoneDetails = contact.phoneNumbers.find(
+      (aPhone: ContactPhoneDetails) => aPhone.contactPhoneId === parseInt(contactPhoneId, 10),
+    )
+    if (!phone) {
+      throw new Error(
+        `Couldn't find phone with id ${contactPhoneId} for contact ${contactId}. URL probably entered manually.`,
+      )
+    }
+    const currentType = res.locals?.formResponses?.type ?? phone.phoneType
```

Comment:
why do we need do this, is it not safe to get it directly from phone? `const currentType = res.locals?.formResponses?.type ?? phone.phoneType`